node/src/actors/chain_manager/handlers.rs

```diff
@@ -89,13 +88,11 @@ impl Handler<EpochNotification<EveryEpochPayload>> for ChainManager {
                 if let (Some(current_epoch), Some(chain_info)) =
                     (self.current_epoch, self.chain_state.chain_info.as_ref())
                 {
-                    let candidates = self.candidates.clone();
-
                     // Decide the best candidate
                     // TODO: replace for loop with a try_fold
                     let mut chosen_candidate = None;
-                    for (key, block_candidate) in candidates {
-                        if let Some((chosen_key, _)) = chosen_candidate.clone() {
+                    for (key, block_candidate) in self.candidates.drain() {
```

Comment:
Now that we are 
Since now we are draining the vector, in line 152 there is a redundant clear:
```rust
// Clear candidates
self.candidates.clear();
```